### PR TITLE
Add empty_trailing_lines check

### PR DESCRIPTION
**Pull request by treydock**

### CODE REVIEW

**Comment by treydock** [comment]
Once merged this is steps to publish (at bottom of page): http://puppet-lint.com/developer/tutorial/#publish-it

**Comment by ghoneycutt** [comment]
Could you please setup travis to auto publish to rubygems. You can see how from the voxpupuli puppet-lint plugins.

File: puppet-lint-empty_trailing_lines.gemspec [hunk 1]
```diff
@@ -0,0 +1,26 @@
+Gem::Specification.new do |spec|
+  spec.name        = 'puppet-lint-empty_trailing_lines'
+  spec.version     = '0.0.1'
+  spec.homepage    = 'https://github.com/ghoneycutt/puppet-lint-empty_trailing_lines'
+  spec.license     = 'Apache-2.0'
+  spec.author      = 'Garret Honeycutt'
+  spec.email       = 'gh@tailoredautomation.io'
```
Comment (ghoneycutt) [inline puppet-lint-empty_trailing_lines.gemspec]: author: `Tailored Automation`
email: `code@tailoredautomation.io`

File: puppet-lint-empty_trailing_lines.gemspec [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+    'lib/**/*',
+    'spec/**/*',
+  ]
+  spec.test_files  = Dir['spec/**/*']
+  spec.summary     = 'puppet-lint empty_trailing_line check'
+  spec.description = <<-EOF
+    Extends puppet-lint to ensure that your manifest files do not end with extra empty lines
+  EOF
+
+  spec.add_dependency             'puppet-lint', '>= 1.0', '< 3.0'
```
Comment (ghoneycutt) [inline puppet-lint-empty_trailing_lines.gemspec]: We should probably just support puppet-lint v2

**Comment by ghoneycutt** [comment]
Also need the testing harness for travis. See https://github.com/voxpupuli/puppet-lint-absolute_classname-check/blob/master/.travis.yml

We should just support rubies for latest puppet 5 and 6 - 2.4.5 and 2.5.7

File: puppet-lint-empty_trailing_lines.gemspec (hunk 1 above)
Comment (treydock) [reply puppet-lint-empty_trailing_lines.gemspec]: Updated

File: puppet-lint-empty_trailing_lines.gemspec (hunk 2 above)
Comment (treydock) [reply puppet-lint-empty_trailing_lines.gemspec]: Updated

**Comment by treydock** [comment]
Added travis.yml based on voxpupuli but testing just puppet versions of Ruby

**Comment by treydock** [comment]
I think you need to log into travis and enable this repo to run travis, or something because I think travis is not hooked up for this repo

File: spec/puppet-lint/plugins/check_empty_trailing_lines_spec.rb [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+  context 'with fix enabled' do
+    before do
+      PuppetLint.configuration.fix = true
+    end
+
+    after do
+      PuppetLint.configuration.fix = false
+    end
+
+    context 'code ending with an extra newline' do
```
Comment (ghoneycutt) [inline spec/puppet-lint/plugins/check_empty_trailing_lines_spec.rb:49]: what happens if there are multiple lines like

`test\n\n\n\n\n`

**Comment by ghoneycutt** [comment]
Activated travis

**Comment by ghoneycutt** [comment]
Not sure if you need rubygems password or api token or what encrypted.

**Comment by treydock** [comment]
Added fix to handle lots of extra newlines via recursive function and added test. Also I don't think authentication is required to publish ruby gems.

**Pull request merged**